packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,383 @@
+// @ts-check
+import { Buffer } from 'buffer';
+import ReadlineTransform from 'readline-transform';
+import { Readable } from 'stream';
+import { assert, Fail, q } from '@agoric/assert';
+import { createSHA256 } from './hasher.js';
+
+/**
+ * @typedef { import('./swingStore').SwingStoreExporter } SwingStoreExporter
+ *
+ * @typedef {{
+ *   initTranscript: (vatID: string) => void,
+ *   rolloverSpan: (vatID: string) => void,
+ *   getCurrentTranscriptBounds: (vatID: string) => { startPos: number, endPos: number, hash: string, size: number },
+ *   writeTranscriptItem: (vatID: string, item: string) => void,
+ *   readTranscript: (vatID: string, startPos?: number, endPos?: number) => IterableIterator<string>,
+ *   closeTranscript: (vatID: string) => void,
+ *   getExportRecords: (includeHistorical: boolean) => Iterable<[key: string, value: string]>,
+ *   getArtifactNames: (includeHistorical: boolean) => Iterable<string>,
+ *   getTranscript: (vatID: string, startPos: number) => AsyncIterable<Uint8Array>
+ *   importTranscript: (artifactName: string, exporter: SwingStoreExporter, artifactMetadata: Map) => Promise<void>,
+ * }} TranscriptStore
+ *
+ * @typedef {{
+ *   dumpTranscripts: (includeHistorical?: boolean) => any,
+ * }} TranscriptStoreDebug
+ *
+ */
+
+function* empty() {
+  // Yield nothing
+}
+
+/**
+ * @param {unknown} position
+ * @returns {asserts position is number}
+ */
+
+function insistTranscriptPosition(position) {
+  assert.typeof(position, 'number');
+  assert(position >= 0);
+}
+
+/**
+ * @param {*} db
+ * @param {() => void} ensureTxn
+ * @param {(key: string, value: string) => void} noteExport
+ * @returns { TranscriptStore & TranscriptStoreDebug }
+ */
+export function makeTranscriptStore(db, ensureTxn, noteExport) {
+  db.exec(`
+    CREATE TABLE IF NOT EXISTS transcriptItems (
+      vatID TEXT,
+      position INTEGER,
```

Comment:
I like `pos` because it's short and composes well with modifiers in the various contexts where it is used.  Per our discussion this afternoon I'm going to leave this one as-is.